go-controller/pkg/ovn/gateway.go

```diff
@@ -526,13 +525,23 @@ func (gw *GatewayManager) GatewayInit(
 				types.NetworkExternalID:  gw.netInfo.GetNetworkName(),
 				types.TopologyExternalID: gw.netInfo.TopologyType(),
 			}
+			if !config.OVNKubernetesFeature.EnableInterconnect {
+				lrsr.ExternalIDs["nodeName"] = nodeName
+			}
+		} else if !config.OVNKubernetesFeature.EnableInterconnect {
```

Comment:
Was it intentional that you override all ExternalIDs in case of non-secondary networks?
Otherwise, You could just extract the if and do it once:
```
if gw.netInfo.IsSecondary() {
...
}
...
if !config.OVNKubernetesFeature.EnableInterconnect {
}

```